ql/patterns/lazyobject.hpp

```diff
@@ -172,6 +177,17 @@ namespace QuantLib {
         }
     }
 
+    /*! Calls fowardFirstNotificationOnly on o and all observables of o that are a lazy object,
+       recursively. This reduces the number of notifications sent from the lazy objects, but might
+       result in missing recalculations of objects observing one of the lazy object directly or
+       indireclty. */
+    inline void enableSingleNotificationFromLazyObjects(ext::shared_ptr<Observer> o) {
+        if (auto l = ext::dynamic_pointer_cast<LazyObject>(o))
+            l->forwardFirstNotificationOnly();
+        for (auto& p : o->observables())
+            if (auto l = ext::dynamic_pointer_cast<Observer>(p))
+                enableSingleNotificationFromLazyObjects(l);
+    }
```

Comment:
I think this might be misleading.  On the one hand, it's going to be invalidated as soon as one of the recursively-observed handles is relinked; the new linked structure won't have the setting applied.  On the other hand, by calling this on an observer we're implicitly and silently changing the behavior for any other observer that happens to watch the same observables.  And on the third hand, I think it makes more sense to make the choice based on the specific observable (e.g., building a piecewise curve based on two dozen quotes? Maybe set the flag.)